spring-integration-jdbc/src/main/java/org/springframework/integration/jdbc/store/channel/MessageGroupPreparedStatementSetter.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2002-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.jdbc.store.channel;
+
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+
+import org.springframework.integration.jdbc.store.JdbcChannelMessageStore;
+import org.springframework.jdbc.core.PreparedStatementSetter;
+import org.springframework.messaging.Message;
+
+/**
+ * Callback to be used with {@link JdbcChannelMessageStore}.
+ * <p>
+ * Behavior is same as standard {@link PreparedStatementSetter}, it takes in
+ * extra {@code Message<?> requestMessage} and {@code Object groupId} as
+ * parameter used for {@code addMessageToGroup} method in
+ * {@link JdbcChannelMessageStore}.
+ *
+ * @author Meherzad Lahewala
+ * @since 5.0
+ * @see PreparedStatementSetter
+ */
+@FunctionalInterface
+public interface MessageGroupPreparedStatementSetter {
+
+	void setValues(PreparedStatement ps, Message<?> requestMessage, Object groupId) throws SQLException;
```

Comment:
I would say that this is very high from the requirements perspective. A target developer indeed must know all the `JdbcChannelMessageStore` internals to implement this properly.

I would suggest an approach similar to the `MessageRowMapper` and definitely populate the required statement fields  in this external class. I would make `serializer` and `lobHandler` as optional properties do not perform those hard operation in the custom implementation and leave the rest of logic for the end-user code.
The `priorityEnabled` should be as one more method argument.
The empty ctor  (with nulls for the `serializer` and `lobHandler`) should be protected to be available only for the implementors.